### ISSUE TITLE
Alarm Config Logger Compile Failure

### ISSUE THREAD

**Issue by tanviash** [shroffk/phoebus#403]
I'm seeing compilation failure for alarm-config-logger with the most recent commit on github.

devuser@debian:~/phoebus/services/alarm-config-logger$ mvn clean install
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$1 (file:/usr/share/maven/lib/guice.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for org.phoebus:service-alarm-config-logger:jar:0.0.1-SNAPSHOT
[WARNING] 'build.plugins.plugin.version' for org.springframework.boot:spring-boot-maven-plugin is missing. @ org.phoebus:service-alarm-config-logger:[unknown-version], /home/devuser/phoebus/services/alarm-config-logger/pom.xml, line 62, column 15
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building service-alarm-config-logger 0.0.1-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ service-alarm-config-logger ---
[INFO] Deleting /home/devuser/phoebus/services/alarm-config-logger/target
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ service-alarm-config-logger ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] Copying 0 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.6.2:compile (default-compile) @ service-alarm-config-logger ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 3 source files to /home/devuser/phoebus/services/alarm-config-logger/target/classes
[INFO] /home/devuser/phoebus/services/alarm-config-logger/src/main/java/org/phoebus/alarm/logging/AlarmConfigLogger.java: /home/devuser/phoebus/services/alarm-config-logger/src/main/java/org/phoebus/alarm/logging/AlarmConfigLogger.java uses or overrides a deprecated API.
[INFO] /home/devuser/phoebus/services/alarm-config-logger/src/main/java/org/phoebus/alarm/logging/AlarmConfigLogger.java: Recompile with -Xlint:deprecation for details.
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/devuser/phoebus/services/alarm-config-logger/src/main/java/org/phoebus/alarm/logging/AlarmConfigLogger.java:[27,32] cannot find symbol
  symbol:   class Consumed
  location: package org.apache.kafka.streams
[ERROR] /home/devuser/phoebus/services/alarm-config-logger/src/main/java/org/phoebus/alarm/logging/AlarmConfigLogger.java:[133,68] cannot find symbol
  symbol:   variable Consumed
  location: class org.phoebus.alarm.logging.AlarmConfigLogger
[ERROR] /home/devuser/phoebus/services/alarm-config-logger/src/main/java/org/phoebus/alarm/logging/AlarmConfigLogger.java:[237,9] method does not override or implement a method from a supertype
[INFO] 3 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2.027 s
[INFO] Finished at: 2018-11-01T14:20:35-04:00
[INFO] Final Memory: 23M/87M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.2:compile (default-compile) on project service-alarm-config-logger: Compilation failure: Compilation failure:
[ERROR] /home/devuser/phoebus/services/alarm-config-logger/src/main/java/org/phoebus/alarm/logging/AlarmConfigLogger.java:[27,32] cannot find symbol
[ERROR] symbol:   class Consumed
[ERROR] location: package org.apache.kafka.streams
[ERROR] /home/devuser/phoebus/services/alarm-config-logger/src/main/java/org/phoebus/alarm/logging/AlarmConfigLogger.java:[133,68] cannot find symbol
[ERROR] symbol:   variable Consumed
[ERROR] location: class org.phoebus.alarm.logging.AlarmConfigLogger
[ERROR] /home/devuser/phoebus/services/alarm-config-logger/src/main/java/org/phoebus/alarm/logging/AlarmConfigLogger.java:[237,9] method does not override or implement a method from a supertype
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

**Comment by kasemir** [comment]
Why didn’t that show in Travis?
Anyway, check changes to alarm-logger service, the confit logger needs same change to import etc because of Kafka client lib update

**Comment by kasemir** [comment]
That explains it.
The error message looks like what I saw in the alarm logger, check the changes in there:
https://github.com/shroffk/phoebus/pull/401/files#diff-58f6eb77455df4d2a03c3fe18a2eae38

`import org.apache.kafka.streams.Consumed` needs to change into `import org.apache.kafka.streams.kstream.Consumed`, and the `punctuate()` method that used to be required no longer exists.

The changes are also mentioned in the Kafka documentation, http://kafka.apache.org/20/documentation.html#upgrade_200_notable, http://kafka.apache.org/20/documentation/streams/upgrade-guide#streams_api_changes_200:
"We moved Consumed interface from org.apache.kafka.streams to org.apache.kafka.streams.kstream as it was mistakenly placed in the previous release. If your code has already used it there is a simple one-liner change needed in your import statement."
